tests/Yaapii.Atoms.Tests/Map/GroupedTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+using Yaapii.Atoms.Enumerable;
+using Yaapii.Atoms.Func;
+using Yaapii.Atoms.IO;
+using Yaapii.Atoms.List;
+using Yaapii.Atoms.Map;
+using Yaapii.Atoms.Number;
+
+namespace Yaapii.Atoms.Map.Tests
+{
+    public sealed class GroupedTest
+    {
+        [Fact]
+        public void GroupsListToDictionaryOfKeysAndLists()
+        {
+            var srcList = new ListOf<string>("ABC", "ABCD", "ABCDE");
+            var keyFunc =
+                new FuncOf<string, double>((str) =>
+                    new NumberOf(
+                        new IO.LengthOf(new InputOf(str)).Value()
+                    ).AsDouble()
+                );
+
+            var valueFunc =
```

Comment:
its not a valueFunc - its the value (in the shape of a func). Func is encoding in names.